client/src/main/java/io/pravega/client/stream/impl/ReaderGroupState.java

```diff
@@ -310,7 +310,7 @@ void update(ReaderGroupState state) {
             if (oldPos != null) {
                 throw new IllegalStateException("Attempted to add a reader that is already online: " + readerId);
             }
-            state.distanceToTail.putIfAbsent(readerId, Long.MAX_VALUE);
+            state.distanceToTail.putIfAbsent(readerId, 10 * ASSUMED_LAG_MILLIS);
```

Comment:
This change does not change the behavior of getRanking() method right, since the distance is always capped to ASSUMED_LAG_MILLIS.

Reply:
It uses a minimum of Assumed_Lag_millis. This value shouldn't matter in practice because it should be quickly replaced by the real value. I just changed it because in my testing it was causing long overflow, which caused values to become negative.

Reply:
> I just changed it because in my testing it was causing long overflow, which caused values to become negative.

I wonder if this is indicating a bug. If it is overflowing, then we are trying to add to the value. Why would we be trying to a value that isn't the "real" value? It looks like we are putting a placeholder value and we are trying to use this value for computations even though it isn't a valid value.

Reply:
It is never used for 'real' I checked the code. It is an artifact of the way the test works. It is injecting segment reassignments without ever updating the distance metric. I have changed the test to act like the normal code, and reverted the change here.